test/Microsoft.TestPlatform.Extensions.HtmlLogger.UnitTests/HtmlLoggerTests.cs

```diff
@@ -0,0 +1,457 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.Extensions.HtmlLogger.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Moq;
+    using ObjectModel = VisualStudio.TestPlatform.ObjectModel;
+    using VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.Extensions.HtmlLogger;
+    using HtmlLoggerConstants = Utility.Constants;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+    using HtmlLogger = VisualStudio.TestPlatform.Extensions.HtmlLogger;
+    using System.Collections.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
+    using System.Runtime.Serialization;
+
+    [TestClass]
+    public class HtmlLoggerTests
+    {
+        private Mock<TestLoggerEvents> events;
+        private Htmllogger htmlLogger;
+        private Dictionary<string, string> parameters;
+        private static string DefaultTestRunDirectory = Path.GetTempPath();
+        private static string DefaultLogFileNameParameterValue = "logfilevalue.trx";
+
+        private Mock<IFileHelper> mockFileHelper;
+        private Mock<XmlObjectSerializer> mockXmlSerializer ;
+        private Mock<IHtmlTransformer> mockHtmlTransformer;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            this.events = new Mock<TestLoggerEvents>();
+            this.mockFileHelper = new Mock<IFileHelper>();
+            this.mockHtmlTransformer = new Mock<IHtmlTransformer>();
+            this.mockXmlSerializer = new Mock<XmlObjectSerializer>();
+            this.htmlLogger = new Htmllogger(this.mockFileHelper.Object, this.mockHtmlTransformer.Object, this.mockXmlSerializer.Object);
+            this.parameters = new Dictionary<string, string>(2);
+            this.parameters[DefaultLoggerParameterNames.TestRunDirectory] = HtmlLoggerTests.DefaultTestRunDirectory;
+            this.parameters[HtmlLoggerConstants.LogFileNameKey] = HtmlLoggerTests.DefaultLogFileNameParameterValue;
+            this.htmlLogger.Initialize(this.events.Object, this.parameters);
+        }
+
+        /// <summary>
+        /// if events is null initialize should throw exception
+        /// </summary>
+        [TestMethod]
+        public void InitializeShouldThrowExceptionIfEventsIsNull()
+        {
+            Assert.ThrowsException<ArgumentNullException>(
+                () =>
+                {
+                    this.htmlLogger.Initialize(null, this.parameters);
+                });
+        }
+
+        /// <summary>
```

Comment:
Add some useful summary for the tests, repeating what's already there in the name of the test is not advisable.